x/mint/internal/keeper/keeper.go

```diff
@@ -97,6 +97,9 @@ func (k Keeper) BondedRatio(ctx sdk.Context) sdk.Dec {
 // MintCoins implements an alias call to the underlying supply keeper's
 // MintCoins to be used in BeginBlocker.
 func (k Keeper) MintCoins(ctx sdk.Context, newCoins sdk.Coins) sdk.Error {
+	if newCoins.Empty() {
```

Comment:
IMHO we should call `IsValid` here instead, no? I can still pass negative or even zero coins here (assume I bypass upstream checks in callers. Furthermore, we should maybe also return an error here if `IsValid` is false.

Reply:
moved the `IsValid` downstream to `supply/internal/keeper/bank.go`

Reply:
I'd leave this as is bc `IsValid` returns `true` on empty coins:
https://github.com/cosmos/cosmos-sdk/blob/74915f1e2a4756a494316e205ef4fee1ef51f4ff/types/coin.go#L199-L204

Reply:
Note `!IsValid()` is checked downstream. I added a `panic` on `Add/SubtractCoins`